internal/runners/packages/searchView.go

```diff
@@ -0,0 +1,245 @@
+	verticalMargin = 7
+	scrollUp       = "up"
+	scrollDown     = "down"
+)
+
+type view struct {
+	width         int
+	height        int
+	content       string
+	searchResults *structuredSearchResults
+	ready         bool
+	viewport      viewport.Model
+}
+
+func NewView(results *structuredSearchResults, out output.Outputer) (*view, error) {
+	outFD, ok := out.Config().OutWriterFD()
+	if !ok {
+		logging.Error("Could not get output writer file descriptor, falling back to stdout")
+		outFD = os.Stdout.Fd()
+	}
+
+	width, height, err := term.GetSize(int(outFD))
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get terminal size")
+	}
+
+	return &view{
+		width:         width,
+		height:        height,
+		searchResults: results,
+	}, nil
+}
+
+func (v *view) Init() tea.Cmd {
+	return nil
+}
+
+func (v *view) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	switch msg := msg.(type) {
+	case tea.MouseMsg:
+		switch msg.Button {
+		case tea.MouseButtonWheelUp:
+			v.viewport.LineUp(3)
+			return v, nil
+		case tea.MouseButtonWheelDown:
+			v.viewport.LineDown(3)
+			return v, nil
+		}
+	case tea.KeyMsg:
+		stringMsg := strings.ToLower(msg.String())
+		switch stringMsg {
+		case "q", "ctrl+c":
+			return v, tea.Quit
+		case "up":
+			v.viewport.LineUp(1)
+			return v, nil
+		case "down":
+			v.viewport.LineDown(1)
+			return v, nil
+		case "pgup":
+			v.viewport.LineUp(v.height - verticalMargin)
+			return v, nil
+		case "pgdown":
+			v.viewport.LineDown(v.height - verticalMargin)
+			return v, nil
+		}
+	case tea.WindowSizeMsg:
+		if !v.ready {
+			// Keep the searching message and command in view
+			v.viewport = viewport.New(msg.Width, msg.Height-verticalMargin)
+			v.content = v.processContent()
+			v.viewport.SetContent(v.processContent())
+			v.ready = true
+		} else {
+			v.width = msg.Width
+			v.height = msg.Height
+		}
+	}
+
+	v.viewport, cmd = v.viewport.Update(msg)
+	cmds = append(cmds, cmd)
+
+	return v, tea.Batch(cmds...)
+}
+
+func (v *view) View() string {
+	return v.viewport.View() + "\n\n" + v.footerView()
+}
+
+func (v *view) processContent() string {
+	maxKeyLength := 0
+	for _, key := range keys {
+		renderedKey := colorize.StyleBold.Render(key)
+		if len(renderedKey) > maxKeyLength {
+			maxKeyLength = len(renderedKey) + 2
+		}
+	}
+
+	doc := strings.Builder{}
+	for _, pkg := range v.searchResults.Results {
+		if pkg.Name != "" {
+			doc.WriteString(formatRow(colorize.StyleBold.Render(keyName), pkg.Name, maxKeyLength, v.width))
+		}
+		if pkg.Description != "" {
+			doc.WriteString(formatRow(colorize.StyleBold.Render(keyDescription), pkg.Description, maxKeyLength, v.width))
+		}
+		if pkg.Website != "" {
+			doc.WriteString(formatRow(colorize.StyleBold.Render(keyWebsite), colorize.StyleCyan.Render(pkg.Website), maxKeyLength, v.width))
+		}
+		if pkg.License != "" {
+			doc.WriteString(formatRow(colorize.StyleBold.Render(keyLicense), pkg.License, maxKeyLength, v.width))
+		}
```

Comment:
Note most of these values should be cyan (see mock).

I've also updated the mock to make the name actionable, so bold and cyan, since it makes sense that we want the name to stand out more than any of the other info.